lib/enigma.rb

```diff
@@ -1,8 +1,40 @@
+require 'date'
 class Enigma
+  def rand_key
+    rand(99999)
+  end
 
-  def pad(number)
-    pad_num = number.to_s
+  def zero_pad(key)
+    pad_num = key.to_s
     pad_num.insert(0, "0") until pad_num.size == 5
     pad_num
   end
+
+  def shift_keys(key)
+    keys = []
+    key.chars.each_with_index do |char, index|
+      next if index == 4
+      keys << char + key[index + 1]
+    end
+    keys
+  end
+
+  def date
+    date = Date.today.strftime("%d%m%y")
+  end
+
+  def produce_offset(date)
+    squared_date = date.to_i * date.to_i
+    squared_date.to_s.chars.last(4)
+  end
```

Comment:
"produce_offset" will take in our properly formatted date, turn it into an integer, square it, turn the result of the square into a string, split that string into an array and then grab the last four characters in that array.